### ISSUE TITLE
[wait on ng] App hangs for invalid URLs

### ISSUE THREAD

**Issue by dunqan** [SAP/spartacus#10540]
Using the wrong URL to open an application breaks it.

Example url:
http://localhost:4200/electronics-spa/en/USD/Brands/Canon/c/brand_10%20or%20(1,2)=(select*from(select%20name_const(CHAR(82,88,106,99,113,78,74,70,73,118,87),1),name_const(CHAR(82,88,106,99,113,78,74,70,73,118,87),1))a)%20--%20and%201%3D1

It also has consequences in SSR, as rendering for this specific URL never finishes, causing it to keep allocated resources forever. 

The error happens when Router option `initialNavigation` is set to `enabled` (which is the default in Spartacus 3.0, can also be set for pre 3.x Spartacus)

**Comment by dunqan** [comment]
There is a workaround, so it's not critical, we need to reevaluate it again, after upgrade to Angular 12.

**Comment by giancorderoortiz** [comment]
Issue happens with Spartacus 3.0 and latest dev branch (June 11)

![Screen Shot 2021-06-11 at 5 43 47 PM](https://user-images.githubusercontent.com/46171897/121751745-bf896300-cadc-11eb-92c4-0b90e7f300fb.png)
![Screen Shot 2021-06-11 at 5 44 49 PM](https://user-images.githubusercontent.com/46171897/121751746-bf896300-cadc-11eb-8e09-4cdd37f869db.png)

**Comment by Xymmer** [comment]
we're wondering if it's because it's select* statement which sounds like intent to hack? :)

**Comment by Xymmer** [comment]
see Stan's comment "There is a workaround, so it's not critical, we need to reevaluate it again, after upgrade to Angular 12."

**Comment by Xymmer** [comment]
we repro'd on demo server which was using 4.1 at the time

**Comment by KateChuen** [comment]
Still happening, with develop branch (installed using the script)
![wrongUrl](https://user-images.githubusercontent.com/45409515/144483734-47ad7624-e20c-4111-a1d6-e62857ad78b4.png)

**Comment by znikola** [comment]
Angular issue to track: https://github.com/angular/angular/issues/44355

**Comment by Platonn** [comment]
The workaround https://github.com/SAP/spartacus/pull/10541/files uses `RouterInitializer` which is the **private API** of Angular.
```
import {
  ɵangular_packages_router_router_h as RouterInitializer,
} from '@angular/router';
```

Note: In ng13 it's still a private API.

We might try to convince the Angular team to expose this class, by describing our use case in a new ticket in their repo.

**Comment by Xymmer** [comment]
conclusion: we reviewed and decided: there's a workaround, it's more angular thing, we'll wait and test when they fix
